src/components/Select/Select.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useMemo,useRef, useState } from 'react';
+
+import { Icon } from '../Icon';
+
+import {
+  SelectButtonWraper,
+  SelectItem,
+  SelectLabel,
+  SelectList,
+  SelectWraper,
+} from './styled';
+import { SelectInterface } from '.';
+
+export const Select = ({
+  label,
+  placeholder,
+  options = [],
+  value,
+  onChange,
+}: SelectInterface) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const selectRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      selectRef.current &&
+      !selectRef.current.contains(event.target as Node)
+    ) {
+      setIsOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const handleOptionClick = (selectedValue: string) => {
+    setIsOpen(false);
+    if (onChange) {
+      onChange(selectedValue);
+    }
+  };
+
+  const renderedLabel = useMemo(
+    () => (label ? <SelectLabel>{label}</SelectLabel> : null),
+    [label],
+  );
+
+  const renderedValue = useMemo(
+    () => (value ? value : placeholder || 'Choose from the list'),
+    [value, placeholder],
+  );
+
+  const renderedOptions = useMemo(
+    () => (
+      <SelectList $isOpen={isOpen}>
```

Comment:
why do you start the name of the prop from `$`?

Reply:
I used prefix $ because i had error like that:

<img width="1440" alt="Снимок экрана 2023-11-29 в 17 02 04" src="https://github.com/NikitOS-1/ChatCloud/assets/69642254/6ba4252e-a2a8-4e00-a3b0-10126229aa16">